sklearn/tree/tree.py

```diff
@@ -269,6 +277,10 @@ def fit(self, X, y, sample_weight=None, check_input=True):
         min_samples_split = max(self.min_samples_split,
                                 2 * self.min_samples_leaf)
 
+        if X_idx_sorted is None and self.presort == True and not issparse(X):
+            X_idx_sorted = np.asfortranarray(np.argsort(X, axis=0),
+                                             dtype=np.int32)
```

Comment:
> Ok, I get the idea is to have "stateless" splitter for multi-threading.
> This should be noted somewhere.

Indeed, I think a note on the code would help future people understanding
the design.